### PR TITLE
bugfix: softmax NaN results caused by large -inf masks

### PR DESCRIPTION
**Pull request by xslingcn**
## 📌 Description

For inputs with many `-inf` masks (e.g. topk masked logits), our current softmax kernel produces all `NaN` results when a thread  sees a slice of input made up entirely of `-inf`s. This pr adds checks to fix it.

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

### CODE REVIEW

File: include/flashinfer/sampling.cuh [hunk 1]
```diff
@@ -359,9 +359,11 @@ __global__ void OnlineSoftmaxFusedKernel(DType* logits, DType* output, DType* te
...
 
     float thread_sum = 0.0f;
+    if (!isinf(block_max)) {
 #pragma unroll
-    for (uint32_t j = 0; j < VEC_SIZE; ++j) {
-      thread_sum += __expf(logits_vec[j] - block_max);  // e^(-inf) is safe to add
+      for (uint32_t j = 0; j < VEC_SIZE; ++j) {
+        thread_sum += __expf(logits_vec[j] - block_max);
+      }
     }
```
[the remark this answers is not in the document]
Comment (xslingcn) [reply include/flashinfer/sampling.cuh]: Do we need to handle inputs made of entirely `-inf`s?

[the remark this answers is not in the document]
Comment (yzh119) [reply include/flashinfer/sampling.cuh]: Yes, when you split vocab, it's possible that the entire chunk have -inf logits.

[the remark this answers is not in the document]
Comment (xslingcn) [reply include/flashinfer/sampling.cuh]: > Yes, when you split vocab, it's possible that the entire chunk have -inf logits.

Splitted chunk of all -inf logits should already be handled by the added checks (in that case this chunk makes no contribution to the running_max and running_denominator so we skip updating them). I think gemini is suggesting adding checks for cases where the entire user input contains only -inf, which I think is probably just UB or can may be rejected on python side?

**Review by yzh119 — APPROVED** [review]
